tests/unit/capi/GEOSGeom_createEnvelopeTest.cpp

```diff
@@ -0,0 +1,33 @@
+// Test Suite for C-API GEOSGeom_createEnvelope
+
+#include <tut/tut.hpp>
+// geos
+#include <geos_c.h>
+// std
+#include <cstdarg>
+#include <cstdio>
+#include <cstdlib>
+
+#include "capi_test_utils.h"
+
+namespace tut {
+struct test_capigeosgeom_createenvelope : public capitest::utility {};
+
+typedef test_group<test_capigeosgeom_createenvelope> group;
+typedef group::object object;
+
+group test_capigeosgeom_createenvelope_group("capi::GEOSGeom_createEnvelope");
+
+template <>
+template <>
+void object::test<1>() {
+    GEOSGeometry* expected = GEOSGeomFromWKT("POLYGON ((0 0, 0 2, 1 2, 1 0, 0 0))");
+    GEOSGeometry* geom = GEOSGeom_createEnvelope(0, 0, 1, 2);
+
+    ensure_equals(GEOSEqualsExact(geom, expected, 0), 1);
+
+    GEOSGeom_destroy(expected);
+    GEOSGeom_destroy(geom);
+}
```

Comment:
Should probably also pin down the behavior for degenerate cases like `createEnvelope(0,0,0,0)`.

Reply:
Meaning, degenerate case should raise an exception & return `NULL`?

Reply:
I don't really have an opinion on what the behavior should be (though it would be easy to just go with what the `GeometryFactory` method produces). I just think it's important to put in tests to avoid unintentional future changes in that behavior, whatever it is.

Reply:
I opted to add validation to the C API side, and raise an error if the bounds do not produce a rectangle with both width and height.

The `GeometryFactory` method produces a point if there is no width and height, which didn't seem right for the return value here.

Reply:
I would opt for not enforcing non-zero width/height.  That doesn't cause any structural problems, it just creates an invalid geometry.  Most (all) GEOS constructors don't enforce validity - they leave that up to the client if it's required.  And this lets the function create extents which can represent point queries.

I think I'd also check the order of the input ordinates and normalize them as required (i.e. swap to ensure min/max order), to produce a rectangle in a known orientation.   Just like the Envelope constructor does.

Reply:
Looks like the Envelope contructor takes care of normalizing the coordinates, so I didn't add anything extra there.

Added note to indicate that this can return a point if width and height are 0.

Reply:
Yes, that's a good way to handle it.